mpd/internal/server/server.go

```diff
@@ -387,6 +387,66 @@ func (s *server) writeResponse(p *textproto.Conn, args []string, okLine string)
 			return
 		}
 		s.currentPlaylist.Add(i)
+	case "prio":
+		if len(args) != 3 {
+			ack("wrong number of arguments")
+			return
+		}
+		rng := strings.Split(args[2], ":")
+		var prio, start, end int
+		var err error
+		prio, err = strconv.Atoi(args[1])
+		if err != nil || prio < 0 || prio > 255 {
+			ack("invalid priority")
+			return
+		}
+
+		if len(rng) == 1 {
+			// Updating a single song from the playlist at position i.
+			i, err := strconv.Atoi(rng[0])
+			if err != nil {
+				ack("invalid song position")
+				return
+			}
+			start = i
```

Comment:
check `start` is less than `s.currentPlaylist.Len()` and >= 0?

Reply:
Checking for start >=0 is sensible, but I don't want to test for s.currentPlaylist.Len() as it doesn't matter for this case.